pkg/apis/cilium/types_network.go

```diff
@@ -161,6 +161,24 @@ type BGPControlPlane struct {
 	Enabled bool
 }
 
+// Encryption enables transparent network encryption for cilium
+type Encryption struct {
```

Comment:
Wireguard is also a supported encryption variant. Keep that in mind in the API, so we should probably have a Encryption.Type field.
Also SecretName, KeyFile and MountPath are only relevant to IPSec.

We should consider to have this API a bit like the helm values for cilium upstream:
https://github.com/cilium/cilium/blob/70ef45a982ef0495575070cd702565a8a4b0c558/install/kubernetes/cilium/values.yaml#L1040C1-L1076C28